### PR TITLE
Remove guarded_addons directory and associated code, migrate existing

### LINKED ISSUE #18869: Stop moving files to `guarded-addons/` and move files already there back to `files/`
Depends on https://github.com/mozilla/addons-server/issues/18844

Once the issue above is done, we'll have full control of permissions for accessing files through the database and we won't need to move disabled/deleted files any longer. We should then:
- Stop moving files to `guarded-addons/`, removing associated `hide_disabled_files` and `unhide_disabled_files` code.
- Move existing files in `guarded-addons/` back to `files/`

### PR DESCRIPTION
**Pull request by diox**
Files are now all kept in the same directory, and `download_file()` view controls access.

This removes the code that was moving the files to the `guarded_addons` directory, fallback code to deal with files not yet moved, and the crons that ensures files were moved if needed.

Fixes #18869

### CODE REVIEW

**Comment by diox** [comment]
Blocked by https://mozilla-hub.atlassian.net/browse/SVCSE-481 ; don't merge

**Review by eviljeff — COMMENTED** [review]
Is it wise to remove all the support for `guarded_addons` _before_ the `migrate_guarded_addons` migration can be run?  

It'll take a non-0 amount of time after the deploy for it to be run and complete and during that time the old files won't be in the right place... right?

File: src/olympia/files/management/commands/migrate_guarded_addons.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+                        remaining_entries=total_entries - entries_migrated,
+                    )
+
+    def migrate_addon(self, addon):
+        start_time = datetime.now()
+        new_addon_path = os.path.join(settings.ADDONS_PATH, addon.name)
+        os.makedirs(new_addon_path, exist_ok=True)
+        files = os.scandir(addon.path)
+        for file_ in files:
+            self.migrate_file(addon, Entry(file_.name, file_.path))
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/files/management/commands/migrate_guarded_addons.py:50]: Yeah, I should probably replace that. I was originally passing `DirEntry` instances but they are annoying to work with in tests.

File: src/olympia/files/management/commands/migrate_guarded_addons.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+class Command(BaseCommand):
+    def handle(self, *args, **options):
+        self.migrate()
+
+    def print_eta(self, *, elapsed, remaining_entries):
+        # total_seconds() keeps microseconds - it should often be 0.xxxxxx in
+        # our case. We use the full precision for computing the ETA, but round
+        # to the second when displaying.
+        total_seconds = elapsed.total_seconds()
+        eta = timedelta(seconds=int(total_seconds * remaining_entries))
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/files/management/commands/migrate_guarded_addons.py:21]: True, that'd be much better. I'll rework it.

**Comment by diox** [comment]
> Is it wise to remove all the support for `guarded_addons` _before_ the `migrate_guarded_addons` migration can be run?
> 
> It'll take a non-0 amount of time after the deploy for it to be run and complete and during that time the old files won't be in the right place... right?

That's fine: it should only affect developers/reviewers downloading unreviewed files or admins looking at disabled files. Given we're just doing hard links (and potentially creating directories) it should be fast enough that it's not so much a problem to break that for the duration of the migration (which should run just after the push).

**Review by eviljeff — APPROVED** [review]
r+ based on "That's fine" and "not so much a problem" 😆

File: src/olympia/files/management/commands/migrate_guarded_addons.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/files/management/commands/migrate_guarded_addons.py]: https://github.com/mozilla/addons-server/pull/19113/commits/13f8cefd81451542f11b18d17871608b2437dc49

**Pull request merged**